hazelcast/src/main/java/com/hazelcast/config/replacer/ExecReplacer.java

```diff
@@ -0,0 +1,138 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.config.replacer;
+
+import static com.hazelcast.util.StringUtil.UTF8_CHARSET;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.Properties;
+
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+
+/**
+ * The {@link ConfigReplacer} implementation which allows to replace variables with a standard output of an executed command.
+ * The prefix of this replacer is {@value #PREFIX} and the full format of variables is
+ * <code>$EXEC{executable arg1 arg2 ...}</code> (e.g. <code>$EXEC{echo Hello}</code>).
+ * <p>
+ * By default whitespaces are used as argument separators. If some argument value contains a whitespace, then configure another
+ * argument separator in {@value #PROPERTY_ARGUMENT_SEPARATOR} init property.
+ */
+public class ExecReplacer implements ConfigReplacer {
+
+    /**
+     * Replacer property name to configure argument separator regular expression.
+     */
+    public static final String PROPERTY_ARGUMENT_SEPARATOR = "argumentSeparator";
+
+    private static final String PREFIX = "EXEC";
+    private final ILogger logger = Logger.getLogger(ExecReplacer.class);
+
+    private String argumentSeparator;
+
+    public void init(Properties properties) {
+        argumentSeparator = properties.getProperty(PROPERTY_ARGUMENT_SEPARATOR, "\\s+");
+    }
+
+    @Override
+    public String getPrefix() {
+        return PREFIX;
+    }
+
+    @Override
+    public String getReplacement(String variable) {
+        try {
+            return processWaitAndGetOutput(variable);
+        } catch (Exception e) {
+            logger.warning("Execution failed for variable " + variable, e);
+        }
+        return null;
+    }
+
+    /**
+     * Waits for termination of the given process and returns Standard output
+     */
+    private String processWaitAndGetOutput(String command) throws InterruptedException, IOException {
+        String[] cmdarray = command.split(argumentSeparator);
+        Process process = Runtime.getRuntime().exec(cmdarray);
+        StdStreamReader outReader = new StdStreamReader(process.getInputStream(), true);
+        outReader.start();
+        StdStreamReader errReader = new StdStreamReader(process.getErrorStream(), false);
+        errReader.start();
+
+        process.waitFor();
+        outReader.join();
+        errReader.join();
+        int exitCode = process.exitValue();
+
+        if (exitCode != 0) {
+            logger.warning("Command finished with non-zero exit code (" + exitCode + "): " + Arrays.toString(cmdarray));
```

Comment:
I would probably go for fail-fast here? as a typo in the command should prevent an instance to start? what's the value of ignoring the non-zero code?